### PR TITLE
qcontainer: fix 'bus_params' referenced before assignment

### PR DESCRIPTION
**Pull request by yanan-fu**
id: 1798985

Signed-off-by: Yanan Fu <yfu@redhat.com>

### CODE REVIEW

**Comment by yanan-fu** [comment]
```
diff --git a/shared/cfg/base.cfg b/shared/cfg/base.cfg
index f946318d..2e602d3b 100644
--- a/shared/cfg/base.cfg
+++ b/shared/cfg/base.cfg
@@ -159,8 +159,18 @@ q35, arm64-pci, arm64-mmio:
 # virtio_port_name_prefix_vs1 = "com.redhat.spice."
 # Then the serial_name of "vs1" will be "com.redhat.spice.0"
 
-serials = "serial0"
-serial_type = "isa-serial"
+serials = "serial0 vs1 vs2 vc1 vc2"
+serial_type_serial0 = "isa-serial"
+serial_type_vs1 = virtserialport
+serial_type_vs2 = virtserialport
+serial_type_vc1 = virtconsole
+serial_type_vc2 = virtconsole
+chardev_backend_vs1 = unix_socket
+chardev_backend_vs2 = tcp_socket
+chardev_backend_vc1 = unix_socket
+chardev_backend_vc2 = tcp_socket
+virtio_serial_extra_params_vs1 = "ioeventfd=on"
+virtio_serial_extra_params_vc1 = "ioeventfd=on"
```
Test with boot, 
```
 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.boot: PASS (43.84 s)
```
qemu command line:
```
    -chardev socket,path=/var/tmp/avocado_geeqoi8q/serial-serial0-20200206-090425-5TtVTxO3,id=chardev_serial0,server,nowait \
    -device isa-serial,id=serial0,chardev=chardev_serial0 \

    -chardev socket,path=/var/tmp/avocado_geeqoi8q/serial-vs1-20200206-090425-5TtVTxO3,id=chardev_vs1,server,nowait \
    -device virtio-serial-pci,ioeventfd=on,id=virtio_serial_pci0,bus=pci.0,addr=0x3 \
    -device virtserialport,id=vs1,name=vs1,chardev=chardev_vs1,bus=virtio_serial_pci0.0,nr=1 \

    -chardev socket,port=5103,id=chardev_vs2,host=127.0.0.1,server,nowait \
    -device virtserialport,id=vs2,name=vs2,chardev=chardev_vs2,bus=virtio_serial_pci0.0,nr=2 \

    -chardev socket,path=/var/tmp/avocado_geeqoi8q/serial-vc1-20200206-090425-5TtVTxO3,id=chardev_vc1,server,nowait \
    -device virtio-serial-pci,ioeventfd=on,id=virtio_serial_pci1,bus=pci.0,addr=0x4 \
    -device virtconsole,id=vc1,name=vc1,chardev=chardev_vc1,bus=virtio_serial_pci1.0,nr=1 \

    -chardev socket,port=5497,id=chardev_vc2,host=127.0.0.1,server,nowait \
    -device virtio-serial-pci,id=virtio_serial_pci2,bus=pci.0,addr=0x5 \
    -device virtconsole,id=vc2,name=vc2,chardev=chardev_vc2,bus=virtio_serial_pci2.0,nr=1  \
```

**Comment by yanan-fu** [comment]
@luckyh please help review, thanks!

**Comment by xutian** [comment]
ACK

File: virttest/qemu_devices/qcontainer.py [hunk 1]
```diff
@@ -2180,11 +2180,10 @@ def serials_define_by_variables(self, serial_id, serial_type, chardev_id,
         """
 
         devices = []
-        if bus_extra_params:
-            bus_params = (dict([_.split('=')
-                                for _ in bus_extra_params.split(',') if _]))
         # For virtio devices, generate controller and create the port device
         if serial_type.startswith('virt'):
+            bus_params = (dict([_.split('=')
+                                for _ in bus_extra_params.split(',') if _]))
```
Comment (luckyh) [inline virttest/qemu_devices/qcontainer.py]: The newly introduced param should work not only for virtio serial controllers, although our tests may not have such requirements currently.

Comment (yanan-fu) [reply virttest/qemu_devices/qcontainer.py]: @luckyh updated, thanks!

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**